CRM/Core/DAO/AllCoreTables.data.php

```diff
@@ -217,6 +217,11 @@
     'class' => 'CRM_Dedupe_DAO_Rule',
     'table' => 'civicrm_dedupe_rule',
   ],
```

Comment:
It would be nice to get rid of `CRM_Dedupe_DAO_Rule` here.
Isn't this file autogenerated? I think that means it would disappear anyway the next time CodeGen is run.

What about adding this to fix APIv3:
```diff
diff --git a/api/v3/utils.php b/api/v3/utils.php
index d96aae6785..5abea6650c 100644
--- a/api/v3/utils.php
+++ b/api/v3/utils.php
@@ -309,6 +309,11 @@ function _civicrm_api3_get_DAO($name) {
     $name = 'Contact';
   }
 
+  // Entity was renamed for APIv4
+  if ($name === 'Rule') {
+    return 'CRM_Dedupe_DAO_DedupeRule';
+  }
+
   // hack to deal with incorrectly named BAO/DAO - see CRM-10859
 
   // FIXME: DAO should be renamed CRM_Mailing_DAO_MailingEventQueue
```

Reply:
Hmm right, yeah thats another place in api3 to indicate the mapping between the old and new entity name. Updated the PR. Thanks for the patch.

Reply:
@colemanw keeping the old dao for a bit reduces the chance we break things as last dao renaming we had a few issues